pages/[id].js

```diff
@@ -130,10 +194,97 @@ export default function PlantDetails({ plants, onDeletePlant, onEditPlant }) {
           isEditMode={true}
         />
       )}
+      <CardDetails>
+        {showPopup && (
+          <PopupContainer>
+            <PopupContent>
+              <h3>Add a Reminder</h3>
+              <p>{plant.name}</p>
+              <label>
+                Task Type:
+                <input
+                  type="text"
+                  value={newTaskType}
+                  onChange={(event) => setNewTaskType(event.target.value)}
+                  required
+                />
+              </label>
+
+              <label>
+                Due Date:
+                <input
+                  type="date"
+                  value={newDueDate}
+                  onChange={(event) => setNewDueDate(event.target.value)}
+                  min={new Date().toISOString().split("T")[0]}
+                  required
+                />
+              </label>
+
+              <label>
+                Repeat Interval:
+                <select
+                  value={repeatInterval}
+                  onChange={(event) => setRepeatInterval(event.target.value)}
+                >
+                  <option value="">Select...</option>
+                  <option value="weekly">Weekly</option>
+                  <option value="bi-weekly">Bi-Weekly</option>
+                  <option value="monthly">Monthly</option>
+                </select>
+              </label>
+
+              <ButtonSave onClick={handleAddReminder}>Add Reminder</ButtonSave>
```

Comment:
Please, use a `form` here, so that you can also make use of `formData`.
```suggestion
<form onSubmit={handleAddReminder}>
              <label>
                Task Type:
                <input
                  type="text"
                  name="taskType"
                  required
                />
              </label>

              <label>
                Due Date:
                <input
                  type="date"
                  name="dueDate"
                  min={new Date().toISOString().split("T")[0]}
                  required
                />
              </label>

              <label>
                Repeat Interval:
                <select name="repeatInterval">
                  <option value="">Select...</option>
                  <option value="weekly">Weekly</option>
                  <option value="bi-weekly">Bi-Weekly</option>
                  <option value="monthly">Monthly</option>
                </select>
              </label>
              <ButtonSave type="submit">Add Reminder</ButtonSave>
</form>
```